File: presentation/src/main/java/info/firozansari/presentation/internal/di/components/ActivityComponent.java

package info.firozansari.presentation.internal.di.components;

import android.app.Activity;
import info.firozansari.presentation.internal.di.PerActivity;
import info.firozansari.presentation.internal.di.modules.ActivityModule;
import dagger.Component;

/**
 * A base component upon which fragment's components may depend.
 * Activity-level components should extend this component.
 *
 * Subtypes of ActivityComponent should be decorated with annotation:
 * {@link info.firozansari.presentation.internal.di.PerActivity}
 */
@PerActivity
@Component(dependencies = ApplicationComponent.class, modules = ActivityModule.class)
interface ActivityComponent {
  //Exposed to sub-graphs.
  Activity activity();
}

File: presentation/src/main/java/info/firozansari/presentation/internal/di/components/ActivityComponent.kt
package info.firozansari.presentation.internal.di.components

import android.app.Activity
import dagger.Component
import info.firozansari.presentation.internal.di.PerActivity
import info.firozansari.presentation.internal.di.modules.ActivityModule

/**
 * A base component upon which fragment's components may depend.
 * Activity-level components should extend this component.
 *
 * Subtypes of ActivityComponent should be decorated with annotation:
 * [info.firozansari.presentation.internal.di.PerActivity]
 */
@PerActivity
@Component(dependencies = [ApplicationComponent::class], modules = [ActivityModule::class])
internal interface ActivityComponent {
    //Exposed to sub-graphs.
    open fun activity(): Activity?
}